framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -529,32 +537,142 @@ public TypeArgumentInference getTypeArgumentInference() {
+     *
+     * <pre>
+     * {@code @Override protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+     *      return getBundledTypeQualifiersWithPolyAll(Regex.class, PartialRegex.class, RegexBottom.class, UnknownRegex.class);
+     *  } }
+     * </pre>
+     *
+     * 4) Support annotations located in a checker's {@literal qual} directory and a
+     * list of other annotations, but without supporting {@link PolyAll}:
+     * <p>
+     *
+     * Place those annotations within the {@literal qual} directory, and override
+     * {@link #createSupportedTypeQualifiers()} by calling
+     * {@link #getBundledTypeQualifiersWithoutPolyAll(Class...)} with a
+     * varargs parameter list of the other annotations. Code example:
+     *
+     * <pre>
+     * {@code @Override protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+     *      return getBundledTypeQualifiersWithoutPolyAll(UnknownFormat.class, FormatBottom.class);
+     *  } }
+     * </pre>
+     *
+     * 5) Supporting only annotations that are explicitly listed:
+     *
+     * Override
+     * {@link #createSupportedTypeQualifiers()} and return an immutable
+     * set of the supported annotations. Code example:
+     *
+     * <pre>
+     * {@code @Override protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+     *      return Collections.unmodifiableSet(
+     *          new HashSet<Class<? extends Annotation>>(
+     *              Arrays.asList(A.class, B.class)));
+     *  } }
+     * </pre>
+     *
+     * The set of qualifiers returned by
+     * {@link #createSupportedTypeQualifiers()} must be an immutable
+     * set. The methods
+     * {@link #getBundledTypeQualifiersWithoutPolyAll(Class...)} and
+     * {@link #getBundledTypeQualifiersWithPolyAll(Class...)} each
+     * return an immutable set.
+     * <p>
+     *
+     * Temporary support for deprecated meta-annotation {@link TypeQualifiers}:
+     * <p>
+     *
+     * If the set is empty, and if the type factory or checker class is
+     * annotated with {@link TypeQualifiers}, return an immutable set with the
+     * same set of classes as the annotation. If the class is not so annotated,
+     * it will return an empty set.
+     * <p>
+     *
+     * @return the type qualifiers supported this processor, or an empty set if
+     *         none
      *
      * @see TypeQualifiers
      */
+    @SuppressWarnings("deprecation")
     protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
-        Class<?> classType;
+        Set<Class<? extends Annotation>> typeQualifiers = new HashSet<Class<? extends Annotation>>();
+        // by default support PolyAll
+        typeQualifiers.addAll(getBundledTypeQualifiersWithPolyAll());
+
+        // temporary support for deprecated @TypeQualifiers annotation
+        // TODO: This support will be removed in the next version of the checker framework
```

Comment:
After this pull request is merged, could you make another that removes support of \@TypeQualifier(s).  This way it will be fresh in your mind.  We will wait to merge it until after the release on Jan 1.